.github/workflows/test-and-deploy.yaml

```diff
@@ -6,7 +6,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: ["3.7", "3.8", "3.9", "3.10", "3.11"]
+        python-version: ["3.8", "3.9", "3.10", "3.11", "3.12"]
```

Comment:
Couple of things to  wrap up to properly drop Python 3.7: 

- CONTRIBUTING.md contains `Version 3.7 or higher is required to run...`
- CONTRIBUTING.md contains `...under multiple Python 3.7+ versions.`
- `mock` dependency from `tests` target should be removed
- `target-version = ["py37"]` → target-version = ["py38"]

We might also specify minimal version in `pyproject.toml` using `requires-python = ">= 3.8"` to make it obvious.